2021/ruby/day1/lib/day1.rb

```diff
@@ -9,13 +11,24 @@ def self.from_file(filepath)
     new(File.foreach(filepath, chomp: true).map(&:to_i))
   end
 
-  def count_increases
+  def answer_part1
     elevations_to_check.each_with_index.reduce(0) do |increases, (elevation, index)|
       previous_elevation = @sonar_report[index]
       increases += (increase?(previous_elevation, elevation) ? 1 : 0)
     end
   end
 
+  def answer_part2
+    (0...elevations_to_check.size).each.reduce(0) do |increases, idx|
+      checked_window = idx...idx + PART2_WINDOW_SIZE
```

Comment:
i took me a while realize `idx + PART2_WINDOW_SIZE` was an expression that evaluated to the right input for the `...` operator. i was trying to figure out how a range of one could be added to an int

Reply:
I like this forward looking windowing technique.

Reply:
@panel - I think I should rename them to `window_start` and `window_end` then put those directly into where I call the indexes. Probably be a little more clear.